parts/windows/kuberneteswindowssetup.ps1

```diff
@@ -188,6 +188,11 @@ $global:TLSBootstrapToken = "{{GetTLSBootstrapTokenForKubeConfig}}"
 # Base64 representation of ZIP archive
 $zippedFiles = "{{ GetKubernetesWindowsAgentFunctions }}"
 
+$useContainerD = ($global:ContainerRuntime -eq "containerd")

```

Comment:
so that's the reason the retry can not work in this case?

Reply:
Could you explain more? For "#2" in the description, I just refined the code to make it more readable.